### PR TITLE
Feature/compile time tsstrict checking

### PR DESCRIPTION
**Pull request by KostkaBrukowa**

### CODE REVIEW

**Comment by KostkaBrukowa** [comment]
Jeszcze readme trzeba dopisać

**Comment by KostkaBrukowa** [comment]
Added readme

File: package.json [hunk 1]
```diff
@@ -24,24 +27,40 @@
     "test": "npm run e2e",
     "build": "tsc",
     "dev": "tsc --watch -p .",
-    "e2e": "node e2e/run.js",
-    "e2e:coverage": "c8 -o e2e_coverage -x e2e -r html -i \"src/**/*\" node e2e/run.js"
+    "e2e": "ts-node e2e/plugin-tests/run.ts && jest",
+    "e2e:coverage": "c8 -o e2e_coverage -x e2e -r html -i \\\\\\\"src/**/*\\\\\\\" node e2e/plugin-tests/run.ts"
```
Comment (kamkry) [inline package.json:31]: why so many backslashes? XD

File: README.md [hunk 2]
```diff
@@ -4,7 +4,12 @@ Typescript plugin that allows turning on strict mode in specific files or direct
 
 ## Do i need this plugin?
 This plugin was created for bigger repositories that want to incorporate typescript strict mode, but project is so big that refactoring everything would take ages. This plugin allows user to simply put `//@ts-strict` comment to a top of a file and turn a strict mode to that file. If needed, strict mode can be turned on to directories too.
-NOTE: this plugin doesn't work in compile time. It will show errors in your IDE but they won't appear during compilation.
+Plugins in general doesn't work in compile time. They will show errors in your IDE but they won't appear during compilation.
+To check strict errors in marked files you can use our script `tsc-strict`.
+This command line tool is created to check for files that should be checked with strict rules in compilation time.
+It finds all files with `//@ts-strict` comment and checks for strict typescript errors only for that files.
```
[the remark this answers is not in the document]
Comment (KostkaBrukowa) [reply README.md:10]: done

File: README.md [hunk 3]
```diff
@@ -51,7 +56,24 @@ Plugin takes one extra non mandatory argument `paths` that is an array of relati
   }
 }
 ```
-All files contained in those paths will be be strictly checked. Yay!
```
[the remark this answers is not in the document]
Comment (KostkaBrukowa) [reply README.md:54]: dont understand

File: README.md [hunk 4]
```diff
@@ -51,7 +56,24 @@ Plugin takes one extra non mandatory argument `paths` that is an array of relati
...
+All files contained in those paths will be strictly checked. Yay!
+
+To add cli tool to your build time you can add a script to scripts list in package.json
+```json
+{
+  "scripts": {
+    ...,
+    "tsc-strict": "tsc-strict"
```
[the remark this answers is not in the document]
Comment (KostkaBrukowa) [reply README.md:66]: done

File: package.json (hunk 1 above)
Comment (KostkaBrukowa) [reply package.json:31]: webstorm magic. Done

**Review by kamkry — APPROVED** [review]

**Pull request merged**